locks/dynamodb/dynamo_lock_item.go

```diff
@@ -108,15 +109,17 @@ func createItemAttributes(itemId string, client *dynamodb.DynamoDB) (map[string]
 	}, nil
 }
 
-// Return the UserID
+// Return the UserName
 func getCallerIdentity(client *dynamodb.DynamoDB) (string, error) {
 	stsconn := sts.New(session.New(), &client.Config)
 	output, err := stsconn.GetCallerIdentity(&sts.GetCallerIdentityInput{})
 	if err != nil {
 		return "", errors.WithStackTrace(err)
 	}
 
-	return *output.UserId, nil
+	userArn := *output.Arn
+	userName := strings.Split(userArn, "/")[1]
```

Comment:
That's a good point. Double checked on this link http://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_variables.html#principaltable

Maybe save the whole ARN now is the safest option.  I'll make the change later today and put test result here.